src/algebra/dual_number.lean

```diff
@@ -0,0 +1,86 @@
+/-
+Copyright (c) 2021 Eric Wieser. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Eric Wieser
+-/
+
+import algebra.triv_sq_zero_ext
+
+/-!
+# Dual numbers
+
+The dual numbers are a special case of `triv_sq_zero_ext R M` with `R = M`, with notation `𝔻[R]` in
```

Comment:
Gotta say, as someone who doesn't recognize the term "dual number," this doesn't explain anything for me :) Could you expand the module doc a bit? Even just repeating the doc string for `dual_number` would help.

Reply:
Done! I feel justified in including the wikipedia link since apparently these aren't familiar to many people (they're used as poor-man's "autodifferentiation" in machine learning).